src/AirshipBindings.iOS.Core/Additions/UAirship.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ Copyright Airship and Contributors
+*/
+
+using System;
+using System.Reflection;
+using Foundation;
+using ObjCRuntime;
+
+namespace UrbanAirship
+{
+    public partial class UAirship
+    {
+        static UAirship()
+        {
+            NSNotificationCenter.DefaultCenter.AddObserver(new NSString("com.urbanairship.airship_ready"), (notification) =>
+            {
+                // Register Airship Xamarin component
+                Version version = Assembly.GetExecutingAssembly().GetName().Version;
+                UAirship.Analytics().RegisterSDKExtension(UASDKExtension.Xamarin, version.ToString());
+            });
+
+        }
+    }
+
+    // SDK-private enum and method declarations
+    [Native]
+    internal enum UASDKExtension : ulong
+    {
+        Cordova = 0,
+        Xamarin = 1,
+        Unity = 2,
+        Flutter = 3,
+        ReactNative = 4
```

Comment:
Since Xamarin assigns fixed integer values to each of the enum values, I think we should do the same in the SDK to avoid any accidental changes through re-ordering or additions. Thoughts?

Reply:
sure, create a ticket

Reply:
https://urbanairship.atlassian.net/browse/MOBILE-1429